pkg/collector/corechecks/containers/containerd/events.go

```diff
@@ -57,23 +113,14 @@ func (s *subscriber) CheckEvents(ctrItf ctrUtil.ContainerdItf) {
 	go s.run(ctx, ev) //nolint:errcheck
 }
 
-func processMessage(id string, message *containerdevents.Envelope) containerdEvent {
-	return containerdEvent{
-		ID:        id,
-		Timestamp: message.Timestamp,
-		Topic:     message.Topic,
-		Namespace: message.Namespace,
-	}
-}
-
 // Run should only be called once, at start time
 func (s *subscriber) run(ctx context.Context, ev containerd.EventService) error {
-	if s.IsRunning() {
+	if s.isRunning() {
 		return fmt.Errorf("subscriber is already running the event listener routine")
 	}
 	stream, errC := ev.Subscribe(ctx, s.Filters...)
 	s.Lock()
-	s.isRunning = true
+	s.running = true
```

Comment:
Since you are touching this code, it appers there is a race here: it is possible for two goroutines to observe `isRunning() == false` and then both will run the code below. This should either use atomic with compare and swap, or the lock needs to be held over the entire check and assign sequence: `lock -> isRunning -> running=true -> unlock`.